### PR TITLE
RT-294 more request body size checks

### PR DESCRIPTION
**Pull request by ismail-s**
- Verify that:
  - attachment descriptions are no longer than 100 chars
  - HL7 payload is no longer than 5MB
  - serialised request body that will get sent to Spine is less than a configurable size, where this is set by default in Terraform to 5MB - 400 bytes. 400 bytes is based on me running one of our integration tests, and getting pycurl to log verbosely, so it logs the HTTP packet it sends. This looked like:
```
POST /reliablemessaging/queryrequest HTTP/1.1
Host: 192.168.128.11
User-Agent: Mozilla/5.0 (compatible; pycurl)
Accept: */*
Accept-Encoding: gzip,deflate
Charset: UTF-8
Soapaction: urn:nhs:names:services:psisquery/QUPC_IN160101UK05
Content-Type: multipart/related; boundary="--=_MIME-Boundary"; type=text/xml; start=ebXMLHeader@spine.nhs.uk
Content-Length: 4498

request-body-here

```
I then looked at how long everything except the request body was and rounded up to 400 bytes.
- Update unittests

### CODE REVIEW

File: mhs/README.md [hunk 1]
```diff
@@ -98,6 +98,11 @@ disabled.
 * `MHS_FORWARD_RELIABLE_ENDPOINT_URL` (outbound only) The URL to communicate with Spine for Forward Reliable messaging
 * `MHS_RESYNC_INITIAL_DELAY` (Outbound service only) The initial delay (in seconds) before making the first poll to the sync-async
     store after the outbound service receives an acknowledgement from Spine
+* `MHS_SPINE_REQUEST_MAX_SIZE` (outbound service only) The maximum size (in bytes) that request bodies sent to Spine
+are allowed to be. This should be set minus any HTTP headers and other stuff in the HTTP packets sent to Spine.
```
Comment (ismail-s) [inline mhs/README.md]: TODO-change other stuff to be sometghing better

File: mhs/common/mhs_common/workflow/common_asynchronous.py [hunk 2]
```diff
@@ -66,10 +68,19 @@ def __init__(self, party_key: str = None, persistence_store: persistence_adaptor
...
 
+        if len(message) > self.max_request_size:
+            logger.error('0007', 'Request to send to Spine is too large after serialisation. '
+                                 '{RequestSize} {MaxRequestSize}',
+                         {'RequestSize': len(message), 'MaxRequestSize': self.max_request_size})
+            await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_PREPARATION_FAILED)
+            return (400, f'Request to send to Spine is too large. MaxRequestSize={self.max_request_size} '
+                         f'RequestSize={len(message)}'), None, None
+
+
```
Comment (ismail-s) [inline mhs/common/mhs_common/workflow/common_asynchronous.py]: TODO-remove this newlin

File: mhs/common/mhs_common/workflow/tests/test_asynchronous_express.py [hunk 3]
```diff
@@ -193,6 +195,25 @@ def test_construct_workflow_with_only_inbound_params(self):
                          self.mock_work_description.set_outbound_status.call_args_list)
         self.mock_transmission_adaptor.make_request.assert_not_called()
 
+    @async_test
+    async def test_handle_outbound_message_fails_with_serialised_message_too_large(self):
```
Comment (ismail-s) [inline mhs/common/mhs_common/workflow/tests/test_asynchronous_express.py:199]: TODO-create jira techdebt for commoning out workflow unittests

File: pipeline/terraform/mhs-environment/variables.tf [hunk 4]
```diff
@@ -226,6 +226,12 @@ variable "mhs_resync_initial_delay" {
   default = 0.150
 }
 
+variable "mhs_spine_request_max_size" {
+  type = number
+  description = "The maximum size of the request body (in bytes) that MHS outbound sends to Spine. This should be set minus any HTTP headers and other stuff in the HTTP packets sent to Spine."
```
Comment (ismail-s) [inline pipeline/terraform/mhs-environment/variables.tf]: TODO-update this description in accordance with previous comment

File: mhs/common/mhs_common/workflow/tests/test_asynchronous_express.py (hunk 3 above)
Comment (ismail-s) [reply mhs/common/mhs_common/workflow/tests/test_asynchronous_express.py:199]: Created [RT-299](https://gpitbjss.atlassian.net/browse/RT-299)

**Review by ghost — APPROVED** [review]

**Pull request merged**